daemon/daemon.go

```diff
@@ -1275,7 +1275,7 @@ func NewDaemon(dp datapath.Datapath) (*Daemon, *endpointRestoreState, error) {
 	// Start watcher for endpoint IP --> identity mappings in key-value store.
 	// this needs to be done *after* init() for the daemon in that function,
 	// we populate the IPCache with the host's IP(s).
-	ipcache.InitIPIdentityWatcher()
+	go ipcache.InitIPIdentityWatcher()
```

Comment:
This creates a race condition, `InitIPIdentityWatcher()` initializes `globalMap` which is referenced in multiple places. You will need to split the init and the starting of the watcher.

Reply:
Fixed